### ISSUE TITLE
FR: Support on Tizen

### ISSUE THREAD

**Issue by JustenDouma** [tailscale/tailscale#10955]
### What are you trying to do?

I want to set up my Samsung smart tv as an exit node, but Tizen is not supported

**Comment by catzkorn** [comment]
Hi,

This is not currently on our product roadmap, but I will leave the feature request open in case that changes.

**Comment by JustenDouma** [comment]
No apps, I wanted to use my TV as an exit node because it's always at home and always connected so it seemed like an obvious and easy option.

**Comment by bradfitz** [comment]
Oh, being an exit node should be possible. We don't need any VPN/tun/etc APIs from Tizen OS to do that.

The main question is whether we can run in the background for long periods of time. It looks like that's maybe possible.

In any case, this isn't at the top of our priority list at the moment. If anybody's interested in working on it, we could maybe sponsor it.

**Comment by Flipfuchs** [comment]
> Is there a particular Tizen app you'd like to be able to access resources on your tailnet? If so, perhaps we could work with that app to embed Tailscale as a library so it can use Tailscale while it's running to access things with Tailscale's userspace network stack, rather than using the OS.

Here with a Samsung TV too, you've asked for a particular App. For me that would be Moonlight, there is no official support for tizen OS but one custom made : https://github.com/OneLiberty/moonlight-chrome-tizen
Would it be possible to enable Tailscale in there?

**Comment by hgabignon** [comment]
Marking my interest as well to leverage a Tizen Jellyfin client with a Jellyfin host available on a Tailnet.

**Comment by a-u19** [comment]
Hi same here want to use it for Jellyfin

**Comment by K0-RR** [comment]
And +1 for an exit node option

**Comment by DragonRollDev** [comment]
And +1 for Jellyfin

**Comment by jmaino** [comment]
I'd like to be able to use another node on my tailnet as an exit node for all traffic from my Samsung TV

**Comment by bradfitz** [comment]
> I'd like to be able to use another node on my tailnet as an exit node for all traffic from my Samsung TV

That seems unlikely to be possible, given the APIs available on Tizen.

**Comment by jowtron** [comment]
It is ironic isn't it? Apple, the company famous for its "walled garden" allows a VPN client, while Samsung, using an open-source Linux base blocks it.

**Comment by bradfitz** [comment]
Yeah, I really do think. It's like a black fly in my Chardonnay.

**Comment by Iceboy-6** [comment]
> > Is there a particular Tizen app you'd like to be able to access resources on your tailnet? If so, perhaps we could work with that app to embed Tailscale as a library so it can use Tailscale while it's running to access things with Tailscale's userspace network stack, rather than using the OS.
> 
> Here with a Samsung TV too, you've asked for a particular App. For me that would be Moonlight, there is no official support for tizen OS but one custom made : https://github.com/OneLiberty/moonlight-chrome-tizen Would it be possible to enable Tailscale in there?

+1

**Comment by perroplanta** [comment]
+1 for Jellyfin

**Comment by dxk-c** [comment]
+1 for Jellyfin, maybe someday we will see it

**Comment by Beraliv** [comment]
+1 for Jellyfin

**Comment by bradfitz** [comment]
Please +1 the top comment to vote; don't leave more comments.

**Comment by bradfitz** [comment]
In any case, I just ordered the smallest Tizen TV I could find for my collection of rando hardware. I'll see what's possible.

**Comment by bradfitz** [comment]
Okay, my Tizen TV arrived. Not sure what we'll do with this yet, but:

<img width="4106" height="3080" alt="Image" src="https://github.com/user-attachments/assets/e237c493-b2ca-4d74-9b59-f5af8ffe09e6" />
<img width="4106" height="3080" alt="Image" src="https://github.com/user-attachments/assets/04f2d21a-c1f0-47e8-b601-a07b5a718dfe" />
<img width="4106" height="3080" alt="Image" src="https://github.com/user-attachments/assets/6d1a6d0b-5ba3-4cc9-a30c-c99f63ea0fdf" />
<img width="4106" height="3080" alt="Image" src="https://github.com/user-attachments/assets/53c29040-fcf9-473c-b30d-bdcb5630e337" />
<img width="4106" height="3080" alt="Image" src="https://github.com/user-attachments/assets/eab93219-aa18-47cc-8519-c515a0194b7d" />

**Comment by jowtron** [comment]
This looks very polished! A few questions:
	•	Is this an official Tailscale build?
	•	How’s the TUN/routing handled, was a privileged API path needed, or is there a userspace workaround?
	•	Any chance this’ll ship via the Samsung store eventually, or is sideload via Tizen Studio the likely path?

Nice to see exit node mode working. The bigger win for most of us would be the TV’s own apps (Plex, Jellyfin etc.) being able to reach servers over the tailnet — is that working too, or is the current build limited to exit node functionality?

**Comment by bradfitz** [comment]
> Is this an official Tailscale build?

No. It was an hour of prototyping. I am an official Tailscale employee, but we don't have an "Official Build" yet.

> How’s the TUN/routing handled, was a privileged API path needed, or is there a userspace workaround?

It's not. This is an exit-node-only app, as seen in the second to last image a bit before it's truncated (I need to make that wrap).

> chance this’ll ship via the Samsung store eventually, or is sideload via Tizen Studio the likely path?

Samsung store might be possible, depending on their review process/rules. Sideload via Tizen Studio would happen sooner.

> The bigger win for most of us would be the TV’s own apps (Plex, Jellyfin etc.) 

Without VPN APIs, I don't see how to make that work without making each of those apps embed Tailscale support.

We could fork the Jellyfin app and add Tailscale support to it, for instance.

**Comment by MulverineX** [comment]
> We could fork the Jellyfin app and add Tailscale support to it, for instance.

This would be great, if the Jellyfin team would accept it.

A comment from you in https://github.com/jellyfin/jellyfin-tizen/issues/409 would be appreciated

Edit: looks like you did comment on there, thanks!

**Comment by bradfitz** [comment]
The Tizen Tailscale exit node app code is at https://github.com/tailscale/tailscale-tizen now.

I agree that it's not super useful, but it serves as a basis/example for people doing things like embedding Tailscale support in a Jellyfin-Tizen app.

**Comment by MulverineX** [comment]
I'll move this here since the Jellyfin issue's discussion has moved in a different direction.

> Yeah, I’m happy to send a PR adding [Tailscale connection to the Jellyfin app] if desired by this repo’s maintainers.
\- bradfitz

> Additionally, this should support spec. compliant 3rd party tailscale/tailnet control servers like https://github.com/juanfont/headscale ...
\- MulverineX

> This kind of functionality is out of scope for a Jellyfin client.
\- nielsvanvelzen (**closed the issue as not planned**)

> @bradfitz if you're still willing to make a fork that would be appreciated, as long as there's a guide to update the tailscale dependencies and update from jellyfin/jellyfin-tizen upstream, the community can probably maintain it from there.
\- MulverineX

**Comment by daydreamer451** [comment]
Hello everyone, 

If my understanding is correct https://github.com/tailscale/tailscale-tizen allows only for the TV to act as a routing exit node. The jellyfin team seems to be against embedding the support for Tailscale into their jellyfin clients/apps. 

Maybe they would be more incline to accept a contribution in the shape of a plugin? Would that be something that is possible to implement? 

Thanks for your time !

**Comment by MulverineX** [comment]
@daydreamer451 chances are they wouldn't allow that either, they have a problem with adding APIs for features they don't personally see fit for the app. If bradfitz is willing to create a fork with the functionality added, then maybe the community can pick up the effort, but the jellyfin team will probably not allow anything like this in any of their official clients.

**Comment by daydreamer451** [comment]
@MulverineX But it seems to me the "third party plugin" ecosystem is community driven and is the space to add additional features to Jellyfin without requiring any sort of approval except compliance with how the plugin has to be setup and the format of the "plugin repository". Which means no need for approval and community members/users can just add and activate the plugin if desired.  Am I misunderstanding something ? Wouldn't that be best bet to avoid upsetting the people at Jellyfin which seem to have other priorities ? 

The sources : https://jellyfin.org/posts/plugin-updates/ & https://github.com/jellyfin/jellyfin-plugin-template

**Comment by MulverineX** [comment]
Those plugins are server-side, not client-side. You're welcome to create a new issue on https://github.com/jellyfin/jellyfin-tizen/ requesting support for a plugin API with the capability to implement something like this in a plugin, but I'm just expressing doubt you'll get anywhere.

**Comment by PatrickSt1991** [comment]
Not sure if anyone is still active on this issue but I like to have a crack at it.

However I don't own a Tizen 8 device and my Tizen 5 is for sure not going to support it.

If anyone has a Tizen 8 and is willing to do some testing;

1. Download Apps2Samsung (https://github.com/Apps2Samsung/Apps2Samsung/releases/tag/v2.7.0-beta)
2. Open the settings view by pressing the cog (gear icon) in the left bottom and toggle this setting;

<img width="622" height="77" alt="Image" src="https://github.com/user-attachments/assets/8f4aa950-7952-47c8-8177-136935c5f816" />

1. Download the .TPK, push it through Apps2Samsung. (Custom WGT File, set filter on .tpk)
https://github.com/PatrickSt1991/tailscale-tizen/releases/tag/tv8-tester

3. Launch app → open About → note the vpnsvc line.

4. If possible: browse http://<your-TV-IP>:8081/  and copy the whole log.

What we're reading:
- vpnsvc: VPNSVC_ERROR_NONE (0) — privilege granted → the partner privilege is live on 8.0+ → I build the kernel-TUN full-tunnel integration.

- vpnsvc: error N — NOT granted → signing/privilege needs fixing before any TUN work.

- The :8081 log also tells us tailscaled actually started + reached Running on their TV (confirms exec isn't blocked on their set, and the exit-node path works there).

That one probe result is the gate for the whole full-tunnel effort. Send over the About line (and :8081 log if possible) and I'll take the next step from there.
